consensus/spos/bls/subroundEndRound.go

```diff
@@ -572,29 +623,29 @@ func (sr *subroundEndRound) verifyNodesOnAggSigFail() ([]string, error) {
 			return nil, err
 		}
 
-		isSuccessfull := true
-		err = sr.SigningHandler().VerifySignatureShare(uint16(i), sigShare, sr.GetData(), sr.Header.GetEpoch())
+		err = sr.checkGoRoutinesThrottler(ctx)
 		if err != nil {
-			isSuccessfull = false
-
-			err = sr.SetJobDone(pk, SrSignature, false)
-			if err != nil {
-				return nil, err
-			}
+			return nil, err
+		}
 
-			// use increase factor since it was added optimistically, and it proved to be wrong
-			decreaseFactor := -spos.ValidatorPeerHonestyIncreaseFactor + spos.ValidatorPeerHonestyDecreaseFactor
-			sr.PeerHonestyHandler().ChangeScore(
-				pk,
-				spos.GetConsensusTopicID(sr.ShardCoordinator()),
-				decreaseFactor,
-			)
+		sr.signatureThrottler.StartProcessing()
 
-			invalidPubKeys = append(invalidPubKeys, pk)
-		}
+		wg.Add(1)
 
-		log.Trace("verifyNodesOnAggSigVerificationFail: verifying signature share", "public key", pk, "is successfull", isSuccessfull)
+		go func(i int, pk string, wg *sync.WaitGroup, sigShare []byte) {
+			defer func() {
+				defer sr.signatureThrottler.EndProcessing()
+				defer wg.Done()
+			}()
```

Comment:
```suggestion
			defer func() {
				sr.signatureThrottler.EndProcessing()
				wg.Done()
			}()
```

they are already in a defer func, no need to defer again